### PR TITLE
Allow respecting PASSWORD_DEFAULT

### PR DESCRIPTION
**Pull request by rullzer**
What do you think about this

@nickvergessen @blizzz @ChristophWurst @georgehrke @juliushaertl ?

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> What do you think about this

makes sense for customization but what is the benefit?

**Comment by rullzer** [comment]
> > What do you think about this
> 
> makes sense for customization but what is the benefit?

On some systems argon2 causes extremely high load.

**Comment by blizzz** [comment]
> > > What do you think about this
> > 
> > 
> > makes sense for customization but what is the benefit?
> 
> On some systems argon2 causes extremely high load.

Usually, you can tune this with the options now exposed. Already enabling a few threads helps. Or even go down on memory on weaker machines.

The core problem is rather to choose a configuration that flies. And admins might not  be knowledgeable in the specifics or shy away from changing security related issues. Offering chossing hashing alghos between those to is rather a "turbo" button. 

Like this it might break with a wrong setting and if PHP is not built with bcrypt.

Ain't it better to implement the Turbo button so that it takes into consideration the available threads and memory and sets the parameters accordingly? Or leave the button and do it in a repair step automatically.

**Comment by rullzer** [comment]
> Usually, you can tune this with the options now exposed. Already enabling a few threads helps. Or even go down on memory on weaker machines.

That doesn't really help with the load. It helps with speed. But if your machine 8 core machine is busy and gets a load of 13 then just having it run with more cores doesn't help.

> The core problem is rather to choose a configuration that flies. And admins might not be knowledgeable in the specifics or shy away from changing security related issues. Offering chossing hashing alghos between those to is rather a "turbo" button.
> 
> Like this it might break with a wrong setting and if PHP is not built with bcrypt.

sure. That is why it says only change if you know what you are doing. We can make the warning more explicit. We have plenty of things that break your nextcloud if you set them wrong.
 
> Ain't it better to implement the Turbo button so that it takes into consideration the available threads and memory and sets the parameters accordingly? Or leave the button and do it in a repair step automatically.

As said I'm not sure I know how to set it in such a way that it helps if your machine is under heavy load already.

**Comment by rullzer** [comment]
Ok alternative suggestion.
We can make a (bool) switch. That says weather or not we should respect PASSWORD_DEFAULT.
That might be more error proof as it is just a bool. And in case of wrong stuff we just set it to false. So less change to brick your nextcloud.

At the same time if php changes the default they willhave a good reason.

**Comment by blizzz** [comment]
If it hammers the server so much, the memory can be capped drastically, this should shorten the runtime and this load, if i got it right. Still, we cannot judge this on setup. 

+1 for PASSWORD_DEFAULT

**Comment by rullzer** [comment]
Moved it to a password default switch

File: config/config.sample.php [hunk 1]
```diff
@@ -1435,6 +1435,16 @@
...
 /**
  * Hashing
+ */
+
+/**
+ * By default Nextcloud will use the Argon2 password hashing if available.
+ * However if for whatever reason you want to stick with the PASSWORD_DEFAULT
+ * of your php version. Then set the setting to true.
+ */
+'hashingDefaultPassword' => false,
```
Comment (ChristophWurst) [inline config/config.sample.php]:
```suggestion
'hashing_default_password' => false,
```

we use snake_case for *most* config params

**Review by ChristophWurst — APPROVED** [review]
Looks good otherwise :+1:

**Comment by blizzz** [comment]
CI is unhappy

**Review by blizzz — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@rullzer 18+ only I would say

**Pull request merged**